src/charm.py

```diff
@@ -129,9 +130,22 @@ def _on_update_status(self, _):
     # END: hook functions
 
     # BEGIN: helper functions
+    def update_external_services(self) -> None:
+        """Attempts to update any external Kubernetes services."""
+        if not self.is_external_client:
+            return
+
+        # every unit attempts to create a bootstrap service
+        # if exists, will silently continue
+        self.node_port_manager.apply_service(
+            service=self.node_port_manager.build_node_port_services(
+                port=Config.MONGOS_PORT
+            )
+        )
```

Comment:
**todo:** Going off of a very limited understanding of `mongos`'s requirements (https://docs.percona.com/percona-operator-for-mongodb/expose.html#service-per-pod), I **think** you need a per-pod `mongos` nodeport service.

Here, pass the 'current' unit/pod in to `build_node_port_services`, and change the created service name to be something like `appname-unitid-external`